### PR TITLE
Replace `ValueError` with `RasaException` in `replace_environment_variables` function

### LINKED ISSUE #7932: read_yaml fails when it tries to expand an environment variable which contains a $
Sentry Issue: [RASA-OPEN-SOURCE-1XP](https://sentry.io/organizations/rasahq/issues/2020529812/?referrer=github_integration)

```
ValueError: Error when trying to expand the environment variables in '${DB_PASSWORD}'. Please make sure to also set these environment variables: '['$dsfdsfwece2s3f']'.
(13 additional frame(s) were not displayed)
...
  File "rasa/utils/endpoints.py", line 26, in read_endpoint_config
    content = rasa.shared.utils.io.read_config_file(filename)
  File "rasa/shared/utils/io.py", line 527, in read_config_file
    content = read_yaml_file(filename)
  File "rasa/shared/utils/io.py", line 368, in read_yaml_file
    return read_yaml(read_file(filename, DEFAULT_ENCODING))
  File "rasa/shared/utils/io.py", line 349, in read_yaml
    return yaml_parser.load(content) or {}
  File "rasa/shared/utils/io.py", line 314, in env_var_constructor
    " variables: '{}'.".format(value, not_expanded)
```

To reproduce:

```
import os
from rasa.shared.utils.io import read_yaml

os.environ['HELLO'] = '$sdfdsf'

print(read_yaml("""
hello:
- ${HELLO}
"""))
```

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- replace `ValueError` with `RasaException` in `replace_environment_variables` function because it's indeed a usage error
- it's basically a post-merge update to #8140
- fixes #7932 

**Status (please check what you already did)**:
- [x] updated tests
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/shared/utils/io.py [hunk 1]
```diff
@@ -314,7 +315,7 @@ def env_var_constructor(loader, node):
             w for w in expanded_vars.split() if w.startswith("$") and w in value
         ]
         if not_expanded:
-            raise ValueError(
+            raise RasaException(
                 "Error when trying to expand the environment variables"
                 " in '{}'. Please make sure to also set these environment"
                 " variables: '{}'.".format(value, not_expanded)
```
Comment (ancalita) [inline rasa/shared/utils/io.py:321]: More of a style comment, shouldn't this be updated to a f-string according to Notion [code convention](https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e#208f2a9207304db8b473ac348a55d26f) ?

Comment (alwx) [reply rasa/shared/utils/io.py:321]: good catch!

**Review by ancalita — COMMENTED** [review]
Looks all good with me 👍 although not sure why the docstring for `read_yaml` mentions a 3rd argument that doesn't seem present in the function signature?

**Comment by alwx** [comment]
@ancalita I guess it's a leftover. I will remove it :)

**Review by ancalita — APPROVED** [review]

**Pull request merged**